modules/apps/questions/questions-web/src/main/java/com/liferay/questions/web/internal/upgrade/v1_1_0/UpgradePortletPreferences.java

```diff
@@ -0,0 +1,73 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2024 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.questions.web.internal.upgrade.v1_1_0;
+
+import com.liferay.message.boards.constants.MBCategoryConstants;
+import com.liferay.message.boards.model.MBCategory;
+import com.liferay.message.boards.service.MBCategoryLocalService;
+import com.liferay.petra.string.StringPool;
+import com.liferay.portal.kernel.portlet.PortletPreferencesFactoryUtil;
+import com.liferay.portal.kernel.upgrade.BasePortletPreferencesUpgradeProcess;
+import com.liferay.portal.kernel.util.GetterUtil;
+import com.liferay.questions.web.internal.constants.QuestionsPortletKeys;
+
+import javax.portlet.PortletPreferences;
+
+/**
+ * @author Marco Galluzzi
+ */
+public class UpgradePortletPreferences
+	extends BasePortletPreferencesUpgradeProcess {
+
+	public UpgradePortletPreferences(
+		MBCategoryLocalService mbCategoryLocalService) {
+
+		_mbCategoryLocalService = mbCategoryLocalService;
+	}
+
+	@Override
+	protected String[] getPortletIds() {
+		return new String[] {QuestionsPortletKeys.QUESTIONS + "%"};
+	}
+
+	@Override
+	protected String upgradePreferences(
+			long companyId, long ownerId, int ownerType, long plid,
+			String portletId, String xml)
+		throws Exception {
+
+		PortletPreferences portletPreferences =
+			PortletPreferencesFactoryUtil.fromXML(
+				companyId, ownerId, ownerType, plid, portletId, xml);
+
+		String value = portletPreferences.getValue("rootTopicId", null);
+
+		if (value != null) {
+			long rootTopicId = GetterUtil.getLong(value);
+
+			String rootTopicExternalReferenceCode = StringPool.BLANK;
+
+			if (rootTopicId != MBCategoryConstants.DEFAULT_PARENT_CATEGORY_ID) {
+				MBCategory mbCategory = _mbCategoryLocalService.fetchMBCategory(
+					rootTopicId);
+
+				if (mbCategory != null) {
+					rootTopicExternalReferenceCode =
+						mbCategory.getExternalReferenceCode();
+				}
+			}
+
+			portletPreferences.setValue(
+				"rootTopicExternalReferenceCode",
+				rootTopicExternalReferenceCode);
+		}
+
+		return PortletPreferencesFactoryUtil.toXML(portletPreferences);
+	}
+
+	private final MBCategoryLocalService _mbCategoryLocalService;
```

Comment:
Can you reverse the conditions? In some cases you're not setting the pref., but in others, you're setting it to null. Better not to set it if there's no value.
```
if (value == null) {
    return PortletPreferencesFactoryUtil.toXML(portletPreferences);
}

...

if (rootTopicId == ...) {
    return PortletPreferencesFactoryUtil.toXML(portletPreferences);
}

...

if (mbCategory == null) {
    return PortletPreferencesFactoryUtil.toXML(portletPreferences);
}

portletPreferences.setValue(..., ...);

return PortletPreferencesFactoryUtil.toXML(portletPreferences);
```

Reply:
Well, I'm setting it the preference to the emtpy string, which has a meaning (no root topic selected). Anyway, I will try to simplify.

Reply:
Done. See last commit [574c892](https://github.com/liferay-content-management/liferay-portal/pull/5927/commits/574c892318435a0934d4a9a824b8ecb6d522ebb8)